code/modules/research/stock_parts/stock_part_datum.dm

```diff
@@ -72,14 +72,129 @@ GLOBAL_LIST_INIT(stock_part_datums, generate_stock_part_datums())
 	physical_object_type = /obj/item/stock_parts/scanning_module
 	physical_object_base_type = /obj/item/stock_parts/scanning_module
 
-/datum/stock_part/scanning_module/tier2
+/datum/stock_part/scanning_module/adv
```

Comment:
Don't do this plz, I made it like this because the flavor names are completely useless

Reply:
But isnt it consistent with the physical object type? Don't see the harm in it

Reply:
it doesn't need to be consistent with the physical object types, which should also be using this kind of naming (though don't change it in this pr). the only reason they might not is because our admin tooling isn't good enough to use names (which i will one day fix)

the harm is that it makes any code using the subtypes less readable. "pico"/"micro"/etc do not mean anything. they are flavor names, and it means i have to go and check what is actually waht

Reply:
Time for search & destroy

Reply:
Ok i wont change